assembly/uastc-rgba32.ts

```diff
@@ -933,22 +959,24 @@ function writeLoopSingleSubsetColorBlock(
 // @ts-ignore: 1206
 @inline
 function writeSingleSubsetColorBlockW5(
-  rl0: i32, rh0: i32, gl0: i32, gh0: i32, bl0: i32, bh0: i32,
-  weights0: i64, weights1: i64,
-  offset: i32, stride: i32): void {
-
-  const rs: i64 = rl0 * 0x40004000400040;
-  const gs: i64 = gl0 * 0x40004000400040;
-  const bs: i64 = bl0 * 0x40004000400040;
-
-  const rd: i64 = rh0 - rl0;
-  const gd: i64 = gh0 - gl0;
-  const bd: i64 = bh0 - bl0;
-
-  const w0: i64 = unqWx4(5, weights0 & 0xFFFFF);
-  const w1: i64 = unqWx4(5, (weights0 >> 20) & 0xFFFFF);
-  const w2: i64 = unqWx4(5, weights1 & 0xFFFFF);
-  const w3: i64 = unqWx4(5, (weights1 >> 20) & 0xFFFFF);
+  rl0: u64, rh0: u64, gl0: u64, gh0: u64, bl0: u64, bh0: u64,
```

Comment:
Before it was:
```ts
  rl0: i32, rh0: i32, gl0: i32, gh0: i32, bl0: i32, bh0: i32,
```
Don't understand why, but if replace all types for this set arguments to `u32` instead `u64`. Tests for `mode = 18` failed. Perhaps need extra check this part.

Reply:
Before:
```ts
let rh0: i32, rl0: i32;
const rd = rh0 - rl0; // may be negative
// casting i32 to i64 preserves value
let r0 = rs + rd * w0 /* i64 + i32 * i64 */
```

After:
```ts
let rh0: u32, rl0: u32;
const rd = rh0 - rl0; // non-negative, wraps in 32 bits
// u32 to u64 cast with wrong value
let r0 = rs + rd * w0 /* u64 + u32 * u64 */
```

Reply:
If `rd` or `r0` does't use sensitive to sign operations like `div`, `mod`, `left shift`, `<`,  `<=`, `>`, `>=`. It doesn't really matter. But I force it to i32/i64 for clarity

Reply:
Seems like `r0` expression is also sign-sensitive although it doesn't use those operations:
```
rh0 = 0, rl0 = 1, w0 = 1;
rs = rl0 * 0x40004000400040 = 0x40004000400040;
```

i32 case:
```
rd = -1 (0xFFFFFFFF)
rd * w0 = <i64>0xFFFFFFFF * 1 = 0xFFFFFFFFFFFFFFFF
r0 = 0x40004000400040 + 0xFFFFFFFFFFFFFFFF = 0x4000400040003F
```

u32 case:
```
rd = 4294967295 (0xFFFFFFFF)
rd * w0 = <u64>0xFFFFFFFF * 1 = 0x00000000FFFFFFFF
r0 = 0x40004000400040 + 0x00000000FFFFFFFF = 0x4000410040003F
```

```
0x4000400040003F != 0x4000410040003F
```

I don't remember why `writeSingleSubsetColorBlockW5` used `i32` instead of `i64` as all other similar functions. Likely an overlook. Thanks for aligning them.

Reply:
Good point. Sign (i64) or zero (u64) extension also matter.